sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -112,6 +133,13 @@ class SQLAppStatusListener(
         }
         exec.jobs = exec.jobs + (event.jobId -> result)
         exec.endEvents += 1
+        // If the jobStart event happened after executionEnd event, remove both stage metrics data
+        // and liveExecution data on the last jobEnd
+        if(exec.endEvents == exec.jobs.size && exec.isStoreReadData) {
```

Comment:
Space after `if`.

Also I'm not sure about this logic. It looks redundant to me, in that it would only ever happen if a job start event arrives after an execution end. It also feels weird to increment `endEvents` here. Don't you get a job end event too?

BTW that's what you use comments for.

Reply:
Modified